CorsixTH/Src/th_pathfind.cpp

```diff
@@ -290,9 +290,15 @@ bool idle_tile_finder::find_idle_tile(const level_map* pMap, int iStartX,
 
   while (true) {
     pNode->visited = true;
-    map_tile_flags flags = pMap->get_tile_unchecked(pNode->x, pNode->y)->flags;
 
-    if (!flags.do_not_idle && flags.passable && flags.hospital) {
+    // Check whether pNode is a feasible destination.
+    const map_tile* node_tile = pMap->get_tile_unchecked(pNode->x, pNode->y);
+    map_tile_flags flags = node_tile->flags;
+    bool correct_parcel = parcelId < 0 || parcelId == node_tile->iParcelId;
```

Comment:
This is where I meant the -1 and 0 'issue' is. I guess its the same reason, that the meander changes are checking for a parcelNr > 0, whereas I would have thought we would have passed it through by default.

Reply:
A long tale to say I should exclude parcel `0` :)

Thanks mugmuggy for spotting this.